design/one-pager-easy-stacks.md

```diff
@@ -0,0 +1,294 @@
+kind: RDSInstanceClass
+metadata:
+  name: dev-project-future-mysql
+  labels:
+    "aws.configurationstacks.crossplane.io/name": project-future
+    "aws.configurationstacks.crossplane.io/namespace": dev
+    "aws.configurationstacks.crossplane.io/uid": 1ca16960-973b-4d58-a6fa-5696638ec631
+specTemplate:
+  writeConnectionSecretsToNamespace: crossplane-system
+  forProvider:
+    dbInstanceClass: db.t2.small
+    masterUsername: root
+    vpcSecurityGroupIDRefs:
+      - name: dev-project-future-rds-security-group
+    dbSubnetGroupNameRef:
+      name: dev-project-future-dbsubnetgroup
+    allocatedStorage: 20
+    engine: mysql
+    skipFinalSnapshotBeforeDeletion: true
+  providerRef:
+    name: aws-provider
+  reclaimPolicy: Delete
+```
+
+Other people in different namespaces can create their own instances of `EasyAWS`
+custom resource and have their own similar environment with different names.
+
+## Technical Implementation
+
+### Metadata of The Stack
+
+[Dependency list] has to be filled with each kind of the resource that the stack
+deploys, i.e. f the stack only deploys `RDSInstanceClass` resource, it should
+not require all AWS stack to be present.
+
+[UI annotations] should be present to make it easy for frontend software to process
+the stack.
+
+### Folder Structure
+
+All directories containing base YAML files should be in a top-level directory
+of the repository named `resources` so that it's easier for others to fork & change.
+
+Controller will scan this folder recursively to get all YAML files. So, the folder
+structure inside that `resources` directory won't matter to the controller but
+it's good to separate the folders cleanly so that it's easier to fork & change.
+
+There are mainly 2 types of resources that configuration stacks will usually
+need to deploy; resource classes and managed resources. Under `resources` folder
+we can have 2 separate folders for each of those resource types; `classes` and
+`services`. Under these folders, resources should be in directories recursively
+named after their group and kind. An example folder structure:
+
+```
+├── resources
+│   ├── classes
+│   │   ├── cache
+│   │   │   └── replicationgroup.yaml
+│   │   ├── compute
+│   │   │   └── ekscluster.yaml
+│   │   └── database
+│   │       ├── rdsinstancemysql.yaml
+│   │       └── rdsinstancepostgresql.yaml
+│   └── services
+│       ├── database
+│       │   └── dbsubnetgroup.yaml
+│       ├── identity
+│       │   ├── iamrole.yaml
+│       │   └── iamrolepolicyattachment.yaml
+│       └── network
+│           ├── internetgateway.yaml
+│           ├── routetable.yaml
+│           ├── securitygroup.yaml
+│           ├── subnet.yaml
+│           └── vpc.yaml
+```
+
+Example above is a suggestion for stacks with one tier of configuration.
+There could also be cases where one configuration stack has the YAML files for
+different tiers and a `spec` allows to deploy one of them, in that case it'd make
```

Comment:
> a `spec` allows to deploy one of them

Hmm, I'm wondering how that would be implemented. But, I'm not sure we need to figure that out before merging this design.

Reply:
It can filter the resources with labels while reading the YAML files.